pkg/plugins/golang/v3/init.go

```diff
@@ -184,9 +184,16 @@ func checkDir() error {
 			if err != nil {
 				return err
 			}
+			// Allow the whole .git directory tree
+			if info.Name() == ".git" {
```

Comment:
If git is being treated specially, why not mercurial/svn?

Reply:
I would say that any '.' prefixed directory/file, `go.mod` and `go.sum` should be allowed.

Reply:
Yes, we need to allow any file/directory with the prefix (.).

Reply:
I proposed this in #1972 but the feedback there was that this wasn’t desirable?

Reply:
```suggestion
			if info.IsDir() && strings.HasPrefix(info.Name(), ".") && info.Name() != "." {
```
Check that it is a directory to avoid returning the skip dir error on non-dirs, checking that the directory starts by `.` and that it is not  exactly `.` (or we will skip everything).